eng/Versions.props

```diff
@@ -8,18 +8,19 @@
     <!-- Opt-in repo features -->
     <UsingToolNetFrameworkReferenceAssemblies>true</UsingToolNetFrameworkReferenceAssemblies>
     <UsingToolSymbolUploader>true</UsingToolSymbolUploader>
-    <MicrosoftBuildVersion>16.7.0</MicrosoftBuildVersion>
-    <MicrosoftBuildTasksCore>16.7.0</MicrosoftBuildTasksCore>
+    <!-- latest versions auto-updated by Maestro/Darc tooling -->
+    <SystemTextJsonVersion>8.0.0-preview.4.23210.22</SystemTextJsonVersion>
+    <!-- Pinned versions -->
```

Comment:
Should we flow the latest msbuild version as well?

Reply:
We technically can't due to MSBuild versions after 16.10 not targeting `netcoreapp3.1` (exception at https://github.com/dotnet/sourcelink/pull/933#discussion_r1162691841). And as far as I am aware of we don't want the two builds diverge, as in we would not want them to use different versions of dependencies

Reply:
The latest officially released MSBuild package supports net7.0. Why can't we use that TFM? Sourcelink's source build is about building the very latest. Why do we care about netcoreapp3.1?

Reply:
Wouldn't we need then 2 MSBuild versions - one for source-build and one for the default build?

Reply:
It should be possible through the `netstandard2.0` ref assemblies we package, yes. Works on my trivial test library . . .

Reply:
Correct, refs to build tasks should be `ExcludeAssets="runtime" PrivateAssets="all"`.

Reply:
> Correct, refs to build tasks should be ExcludeAssets="runtime" PrivateAssets="all".

@rainersigwald thanks. Last question just to double check, .NET Framework compiled tasks should bundle the MSBuild assemblies with it, right?

@oleksandr-didyk based on the above, we should be able to use the latest MSBuild packages just fine.

Reply:
Don't bundle MSBuild assemblies in any TF--any time a task is run, MSBuild will force the use of its own assemblies so they never need to be redistributed.

Reply:
While looking into this more closely I noticed that sourcelink depends on the "Microsoft.Build" package because they us evaluation helper functions which aren't available in the other packages: https://github.com/dotnet/sourcelink/blob/40a5327a26ef250235be54143138325aedd35b79/src/Microsoft.Build.Tasks.Git/GitOperations.cs#L267

That's why libraries need to upgrade from netcoreapp3.1 to net6.0 to use the latest available msbuild package in SBRP: 17.3.2. I submitted https://github.com/dotnet/sourcelink/pull/1001 to fix that.

Reply:
FWIW, sourcelink can't go beyond 17.3.2 because the latest Microsoft.Build doesn't support net6.0 anymore. Even though I chatted with Rainer about this in other threads some time ago, msbuild's package TFM support matrix still confuses me. Does the API that sourcelink uses (`ProjectCollection.Escape`) justify the locked dependency?

Reply:
> I submitted https://github.com/dotnet/sourcelink/pull/1001 to fix that.

Thanks! I will wait for that bump to flow then, unless a different decision around MSBuild's version arises in the meantime

Reply:
@oleksandr-didyk https://github.com/dotnet/sourcelink/pull/1001 is merged and this PR should now be unblocked.